src/test/java/picard/sam/FilterSamReadsTest.java

```diff
@@ -217,7 +218,45 @@ public void testPairedIntervalFilter(final String intervalFilename, final int ex
         Assert.assertEquals(count, expectNumber);
     }
 
-    private FilterSamReads setupProgram(final File inputFile, final File inputSam, final FilterSamReads.Filter filter) throws Exception {
+    @DataProvider(name = "dataTestDebugOption")
+    public Object[][] dataTestDebugOption() {
+        return new Object[][]{
+                {null, false},
+                {true, true},
+                {false, false}
+        };
+    }
+
+    /**
+     * makes sure debug files are created properly
+     */
+    @Test(dataProvider = "dataTestDebugOption")
+    public void testDebugOption(Boolean writeDebugReads, Boolean isDebugFileExpected) throws Exception {
```

Comment:
can't you take in a boolean and make the code simpler?

Reply:
boolean cant be null and I want to test for the default case (the 2nd one could be boolean but it looked weird when I had it that way).  Happy to change the 2nd input though